addons/website_sale/models/res_config_settings.py

```diff
@@ -83,6 +81,21 @@ def _compute_account_on_checkout(self):
         for record in self:
             record.account_on_checkout = record.website_id.account_on_checkout or 'disabled'
 
+    @api.depends('website_id')
+    def _compute_checkout_process_steps(self):
+        """
+        Computing the extra info step and buy now settings when changing
+        the website in the res.config.settings page to show the correct value
+        in the checkbox.
+        """
+        for record in self:
+            record.enabled_extra_checkout_step = record.with_context(website_id=self.website_id.id).website_id.viewref(
```

Comment:
if there is a loop over self, it probably makes sense to use `record` and not `self` in the loop

```suggestion
            record.enabled_extra_checkout_step = record.with_context(website_id=record.website_id.id).website_id.viewref(
```
(it would probably not be an issue here in a res.config.settings but it's better to be cleaner)

same thing 2 lines below